{% extends "media/html/_public.html" %}

{% block content %}
  <div class="row">
    <div class="col-md-6">
      <div class="panel panel-primary">
        <div class="panel-heading">
          <h2 class="panel-title"><span class="glyphicon glyphicon-globe"></span>&nbsp;&nbsp;Primerize Server</h2>
        </div>
        <div class="panel-body">
        <img src="/site_media/images/logo_primerize.png" style="clear:both;float:right;" height="128"/>
          <p class="cap">
            Custom RNA synthesis is of high demand in biological and biotechnology research. Compared to chemical synthesis of RNA, strategies that leverage primer assembly of DNA templates and subsequent <i>in vitro</i> transcription by T7 RNA polymerase are rapid and cost-effective, and RNA lengths up to hundreds of nucleotides are readily achievable. Creating RNAs via this route requires preparing DNA templates, which can be assembled at low cost from mixtures of short primers with lengths up to 60 nucleotides via the polymerase chain reaction (PCR). 
          </p>
          <p>
            This problem can be challenging particularly if one wishes to avoid primer 3´ ends that might misprime into incorrect locations and be extended by DNA polymerase into undesired products, and if one is not allowed to change the sequence. On one hand, there has been substantial work on developing algorithms for designing primers for PCR assembly into DNA templates, with special methods to make codon adjustments for protein synthesis, to optimize primer boundaries against incorrect hybridization of primers, and to assemble large genes. On the other hand with the terminated support of previous web servers, automated primer design tools that optimize against mispriming still require software download, installation, and time to learn.
          </p>
          <p class="bgimg-lg">
            We previously developed a dynamic programming-based algorithm to design primers that can be PCR-assembled into templates for high-throughput RNA synthesis and simple kit or bead-based purification. Given a desired DNA template sequence, this method, named Primerize herein, is optimized to reduce mispriming during PCR by avoiding primer boundaries that might anneal to incorrect sequences. The algorithm has been tested in the synthesis and rapid purification of numerous RNA sequences from our lab with lengths up to 300 nucleotides. In each of these applications, the sequence and purity of the transcribed RNA was verified by reverse transcription and capillary electrophoresis methods. Nevertheless, these scripts previously required MATLAB installation to run and nontrivial efforts to set up. Requests to use this algorithm and the lack of other primer design servers motivated us to prepare an online version of Primerize that should be more broadly useable by the RNA community. 
          </p>
        </div>
      </div>

      <div class="panel panel-danger" id="privacy">
        <div class="panel-heading">
          <h2 class="panel-title"><span class="glyphicon glyphicon-bullhorn"></span>&nbsp;&nbsp;Disclaimer</h2>
        </div>
        <div class="panel-body">
          <h4><span class="label label-orange"><span class="glyphicon glyphicon-eye-close"></span>&nbsp;&nbsp;Privacy</span></h4>
          <p>
            Users of the server are anonymized. Please note that input data and result on the servers are not encrypted by sessions, thus anyone with the <span class="label label-violet">JOB_ID</span> may download a particular result within expiration (9 months). However, the chance of guessing the randomly generated <span class="label label-violet">JOB_ID</span> of any particular result is relatively small. 
          </p>
          <h4><span class="label label-green"><span class="glyphicon glyphicon-bell"></span>&nbsp;&nbsp;Disclosure</span></h4>
          <p>
            No relevant financial relationship or commercial interest with <a href="http://www.idtdna.com" target="_blank">Integrated DNA Technologies (IDT)&nbsp;&nbsp;<span class="glyphicon glyphicon-new-window"></span></a> exists.
          </p>
        </div>
      </div>
    </div>

    <div class="col-md-6">
      <div class="panel panel-success" >
        <div class="panel-heading">
          <h2 class="panel-title"><span class="glyphicon glyphicon-indent-left"></span>&nbsp;&nbsp;Primerize</h2>
        </div>
        <div class="panel-body">
          <p class="cap">
           Primerize (previously named <code>NA_thermo</code>), is an in-house set of MATLAB scripts developed by the Das Lab. It is a small archive of primer design and nucleic acid thermodynamic scripts used by the Das lab at Stanford for high-throughput RNA synthesis and design.
          </p>
          <p>
            The algorithm designs forward (sense strand) and reverse (anti-sense strand) primers that minimize the total length, and therefore the total synthesis cost, of the oligonucleotides. Although developed independently, Primerize is a special case of the general <b>‘Gapped Oligo Design’</b> algorithm described and mathematically analyzed in detail by <a href="http://ieeexplore.ieee.org/xpls/abs_all.jsp?arnumber=4375554" target="_blank">Thachuk and Condon&nbsp;&nbsp;<span class="glyphicon glyphicon-new-window"></span></a>, optimizing the mispriming score and sequence spanning instead of T<sub>m</sub>.
          </p>
          <p>
            We reasoned that mis-hybridizations would occasionally occur and be recognized by DNA polymerase, and since the primer’s 3´-most nucleotide would be bound to the complement, the products could be extended by the polymerase. Thus we introduced and experimentally tested a score term misprime, whose purpose was to reduce artifacts from mispriming during PCR assemblies. A heuristic score was defined based on the number of contiguous reverse complement matches of the 3´-most n nucleotides of the primer on a possible touchdown site; the score was incremented by 10 for A/T pairings and 12.5 for G/C pairings for each match. 
          </p>
        </div>
      </div>
      <div class="panel panel-info" id="citation">
        <div class="panel-heading">
          <h2 class="panel-title"><span class="glyphicon glyphicon-education"></span>&nbsp;&nbsp;Citation</h2>
        </div>
        <div class="panel-body">
          <b>Please cite use of Primerize Server with:</b>
          <br/>
          <blockquote class="h5 alert alert-orange" style="line-height: 140%; font-weight: normal">
            <p>
              Tian S., <i>et al.</i> (<b>2015</b>) <a href="http://nar.oxfordjournals.org/content/43/W1/W522" target="_blank"><u>Primerize: automated primer assembly for transcribing interesting RNAs.</u>&nbsp;&nbsp;<span class="glyphicon glyphicon-new-window"></span></a> <i>Nucleic Acid Research</i> <b>43(W1)</b>: W522-W526.
            </p>
          </blockquote>
          <b>Publications of <u>mutate-map-rescue</u> pipeline:</b>
          <br/>
          <blockquote class="h5" style="line-height: 140%; font-weight: normal">
            <p>
              Tian S., <i>et al.</i> (<b>2014</b>) <a href="http://rnajournal.cshlp.org/content/20/11/1815" target="_blank"><u>High-throughput mutate-map-rescue evaluates SHAPE-directed RNA structure and uncovers excited states.</u>&nbsp;&nbsp;<span class="glyphicon glyphicon-new-window"></span></a> <i>RNA</i> <b>20(11)</b>: 1815-1826.
            </p>
            <p>
              Kladwang W., <i>et al.</i> (<b>2011</b>) <a href="http://www.nature.com/nchem/journal/v3/n12/full/nchem.1176.html" target="_blank"><u>A two-dimensional mutate-and-map strategy for non-coding RNA structure.</u>&nbsp;&nbsp;<span class="glyphicon glyphicon-new-window"></span></a> <i>Nature Chemistry</i> <b>3</b>: 954-962.
            </p>
          </blockquote>
          <b>Other publications using Primerize design:</b>
          <br/>
          <blockquote class="h5" style="line-height: 140%; font-weight: normal">
            <p>
              Cheng C., <i>et al.</i> (<b>2015</b>) <a href="http://elifesciences.org/content/early/2015/06/02/eLife.07600" target="_blank"><u>Consistent and blind inference of RNA tertiary folds through multidimensional chemical mapping.</u>&nbsp;&nbsp;<span class="glyphicon glyphicon-new-window"></span></a> <i>eLife</i> <b>in press, epub available</b>.
            </p>
            <p>
              Miao Z., <i>et al.</i> (<b>2015</b>) <a href="http://rnajournal.cshlp.org/content/21/6/1066" target="_blank"><u>RNA-Puzzles Round II: assessment of RNA structure prediction programs applied to three large RNA structures.</u>&nbsp;&nbsp;<span class="glyphicon glyphicon-new-window"></span></a> <i>RNA</i> <b>21(6)</b>: 1066-1084.
            </p>
            <p>
              Kladwang W., <i>et al.</i> (<b>2014</b>) <a href="http://pubs.acs.org/doi/abs/10.1021/bi5003426" target="_blank"><u>Standardization of RNA Chemical Mapping Experiments.</u>&nbsp;&nbsp;<span class="glyphicon glyphicon-new-window"></span></a> <i>Biochemistry</i> <b>53(19)</b>: 3063-3065
            </p>
            <p>
              Lee J., <i>et al.</i> (<b>2013</b>) <a href="http://www.pnas.org/content/111/6/2122" target="_blank"><u>RNA design rules from a massive open laboratory.</u>&nbsp;&nbsp;<span class="glyphicon glyphicon-new-window"></span></a> <i>Proceedings of the National Academy of Sciences</i> <b>111(6)</b>: 2122-2127.
            </p>
            <p>
              Cordero P., <i>et al.</i> (<b>2012</b>) <a href="http://pubs.acs.org/doi/abs/10.1021/bi3008802" target="_blank"><u>Quantitative Dimethyl Sulfate Mapping for Automated RNA Secondary Structure Inference.</u>&nbsp;&nbsp;<span class="glyphicon glyphicon-new-window"></span></a> <i>Biochemistry</i> <b>51(36)</b>: 7037-7039.
            </p>
          </blockquote>
        </div>
      </div>
    </div>
  </div>

  <div class="row">
    <div class="col-md-12">
      <div class="panel panel-warning" id="contact">
        <div class="panel-heading">
          <h2 class="panel-title"><span class="glyphicon glyphicon-send"></span>&nbsp;&nbsp;Contact</h2>
        </div>
        <div class="panel-body">
          <div class="col-md-1"></div>
          <div class="col-md-2">
            <h4 style="line-height: 140%;">
              <span class="label label-primary pull-right">Principal Investigator</span>
              <br/>
              <span class="label label-success pull-right">Server Team</span>
            </h4>
          </div>
          <div class="col-md-3">
            <h4 style="line-height: 140%;">
              <span class="label label-inverse pull-left">Rhiju Das</span><a href="mailto:rhiju@stanford.edu" target="_blank"><span class="glyphicon glyphicon-envelope pull-right"></span></a>
              <br/>
              <span class="label label-inverse pull-left">Siqi Tian</span><a href="mailto:t47@stanford.edu" target="_blank"><span class="glyphicon glyphicon-envelope pull-right"></span></a>
              <br/>
              <span class="label label-default pull-left">Joseph Yesselman</span>
              <br/>
              <span class="label label-default pull-left">Pablo Cordero</span>
            </h4>
          </div>
          <div class="col-md-6">
            <br/>
            <p>Please feel free to Contact the Das Lab for further questions/issues regarding the web server, suggestions for improvements, citations, or commercial use authorization.</p>
          </div>
        </div>
      </div>
      <div class="panel panel-default" id="history">
        <div class="panel-heading">
          <h2 class="panel-title"><span class="glyphicon glyphicon-time"></span>&nbsp;&nbsp;History</h2>
        </div>
        <div class="panel-body">
          <div class="row">
            <div class="col-md-1"></div>
            <div class="col-md-10">
              <table class="table table-hover">
                <thead>
                  <tr class="active">
                    <th class="col-md-3"><span class="glyphicon glyphicon-calendar"></span>&nbsp;&nbsp;Date</th>
                    <th class="col-md-9"><span class="glyphicon glyphicon-volume-up"></span>&nbsp;&nbsp;Event</th>
                  </tr>
                </thead>
                <tbody>
                  {% for hist in history %}
                    <tr>
                      <td><i>{{hist.date}}</i></td>
                      <td>{% autoescape off %}{{hist.content_html}}{% endautoescape %}</td>
                    </tr>
                  {% endfor %}
                  <tr><td colspan="2" style="padding: 0px;"></td></tr>
                </tbody>
              </table>
            </div>
            <div class="col-md-1"></div>
          </div>
        </div>
      </div>

    </div>
  </div>
{% endblock %}

